packages/core/src/utils.ts

```diff
@@ -331,3 +339,39 @@ export class RateLimitter {
     await await Promise.race(this.processingQueue.values())
   }
 }
+
+/**
+ * @public
+ */
+export interface IndexedReader {
```

Comment:
Do we still need it here?

Reply:
You are right, not here, will move it to the only place where I'm using it